Paco-Server/src/com/google/sampling/experiential/server/CloudSqlSearchServlet.java

```diff
@@ -0,0 +1,100 @@
+package com.google.sampling.experiential.server;
+
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.io.UnsupportedEncodingException;
+import java.sql.SQLException;
+import java.util.List;
+import java.util.logging.Logger;
+
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.joda.time.DateTimeZone;
+import org.json.JSONException;
+
+import com.google.appengine.api.users.User;
+import com.google.sampling.experiential.shared.EventDAO;
+import com.pacoapp.paco.shared.model2.SQLQuery;
+import com.pacoapp.paco.shared.util.QueryJsonParser;
+import com.pacoapp.paco.shared.util.SearchUtil;
+
+import net.sf.jsqlparser.JSQLParserException;
+import net.sf.jsqlparser.statement.select.PlainSelect;
+
+@SuppressWarnings("serial")
+public class CloudSqlSearchServlet extends HttpServlet {
+  public static final Logger log = Logger.getLogger(CloudSqlSearchServlet.class.getName());
+
+  @Override
+  public void doGet(HttpServletRequest req, HttpServletResponse resp) throws IOException, ServletException {
+    throw new ServletException("Method not supported");
+  }
+
+  @Override
+  public void doPost(HttpServletRequest req, HttpServletResponse resp) throws IOException, ServletException {
+    setCharacterEncoding(req, resp);
+    User user = AuthUtil.getWhoFromLogin();
+    String loggedInUser = null;
+    if (user == null) {
+      AuthUtil.redirectUserToLogin(req, resp);
+    } else {
+      loggedInUser = AuthUtil.getEmailOfUser(req, user);
+      SQLQuery sqlQueryObj = null;
+      List<EventDAO> evtList = null;
+      String aclQuery = null;
+      PlainSelect ps = null;
+
+      PrintWriter out = resp.getWriter();
+      resp.setContentType("text/plain");
+
+      CloudSQLDao impl = new CloudSQLDaoImpl();
+      String reqBody = RequestProcessorUtil.getBody(req);
+      try{
+        sqlQueryObj = QueryJsonParser.parseSqlQueryFromJson(reqBody);
+      } catch (JSONException jsonEx) {
+        throw new ServletException("Json parsing error"+jsonEx);
```

Comment:
We should return using EventQueryResultPair. It is setup to be json-ified and it already has semantics for returning errors with an http 200 and an error message. This allows the clients to process errors more effectively. 

You can see an example of its usage in EventServlet.dumpEventsJson on line 168.